File: BorderElementsAdditionMatrix/Add.py
"""
    Author : Pakshal Shashikant Jain
    Date : 26/08/2024
    Program : Write a program which accept matrix from user and return addition of Border
              elements.
"""

#Required libraries
import numpy as np 

############################################################################################################################################33333

#Function to add Border elements of matrix
def AddBorder(arr,irow,icol) :
    i = 0;
    j = 0;
    isum = 0;

    #logic to add Border elements
    for i in range(irow) :
        for j in range(icol) :
            if((i == 0)or(j == 0)or(i == (irow - 1))or(j == (icol - 1))) :
                isum = isum + arr[i][j];

    return isum;

##############################################################################################################################333

#Main Function
def main() :
    i = 0;
    j = 0;
    iRow = 0;
    iCol = 0;
    iret = 0;

    arr = [[]];

    print("Jay Ganesh.....");

    #Taking input from user
    print("Enter number of Rows : ",end = "");
    iRow = int(input());

    print("Enter number of Columns : ",end = "");
    iCol = int(input());

    #Allocating memory of matrix
    arr = np.zeros((iRow,iCol));

    #Taking elements of matrix as input from user 
    for i in range(iRow) :
        print("Enter element of %d row : "%(i +1));
        for j in range(iCol) :
            print("Enter element no %d : "%(j + 1),end = "");
            arr[i][j] = int(input());

    #Printing elements of matrix
    print("Elemnts of matrix :");
    for i in range(iRow) :
        for j in range(iCol) :
            print("%d \t"%arr[i][j],end = "");
        print("");

    #Call to AddBorder function
    iret = AddBorder(arr,iRow,iCol);
    
    #Printing addition of Border elements
    print("Addition of Border elements is : %d"%iret);

##################################################################################################################################################33

#Call back method to call main function
if __name__ == "__main__" :
    #Call to main function
    main();

File: BorderElementsAdditionMatrix/Add.go
/*
    Author : Pakshal Shashikant Jain
    Date : 26/08/2024
    Program : Write a program which accept matrix from user and return addition of Border
              elements.
*/

//Required Libraries
package main;
import "fmt"

//Function to add Border elements
func AddBorder(arr[][] int,irow int,icol int) int {
	var isum int = 0;
	var i int = 0;
	var j int = 0;

	//Logic to add Border elements of matrix
	for i = 0;i < irow;i++ {
		for j = 0;j < icol;j++ {
            if((i == 0)||(j == 0)||(i == (irow - 1))||(j == (icol - 1))) {
                isum = isum + arr[i][j];
            }
		}
	}

	//Returning sum of Border elements
	return isum;
}

//Main Function
func main() {
	var arr[][] int = nil;
	var rows int = 0;
	var cols int = 0;
	var i int = 0;
	var j int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");

	//Taking input from user
	fmt.Print("Enter Number of Rows : ");
	fmt.Scan(&rows);

	fmt.Print("Enter Number of Columns : ");
	fmt.Scan(&cols);

	//Allocating memory for matrix
	arr = make([][]int,rows);

	for i = 0;i < rows;i++ {
		arr[i] = make([]int,cols);
	} 
	
	//Inserting elements in matrix 
	fmt.Println("Enter elements of matrix : ");
	for i = 0;i < rows;i++ {
		fmt.Printf("---Enter elements of %d row : ---\n",(i + 1));
		for j = 0;j < cols;j++ {
			fmt.Printf("Enter element no %d : ",(j + 1));
			fmt.Scan(&arr[i][j]);
		}
	}

	//Printing elements of matrix
	fmt.Println("Elements of matrix are : ");
	for i = 0;i < rows;i++ {
		for j = 0;j < cols;j++ {
			fmt.Printf("%d\t",arr[i][j]);
		}
		fmt.Printf("\n");
	}

	//Call to AddBorder function
	iret = AddBorder(arr,rows,cols);

	//Printing addition of Border elements
	fmt.Println("Addition of Border elements is : ",iret);
}